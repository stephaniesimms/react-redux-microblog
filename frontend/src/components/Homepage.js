import React, { Component } from 'react';
import TitleList from '../containers/TitleList';

/** Homepage: displays list of all blog posts */
class Homepage extends Component {
  render() {
    return (
      <main>
        <p>Welcome to <b>Microblog</b>, a fake forum for heavy metal-themed
        culinary delights. Here you will find diabolical dishes, forbidden fruits, and demonic desserts to exorcise your taste buds. Remember, gluttony is the best sin. Post your devil-icious dishes and leave comments. Vote for the posts you deem the most diabolical. 
        </p>
        <TitleList />
      </main>
    );
  }
}

export default Homepage;