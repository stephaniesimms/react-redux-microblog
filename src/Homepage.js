import React, { Component } from 'react';
import PostList from './PostList';

class Homepage extends Component {
  render() {
    return (
      <PostList />
    );
  }
}

export default Homepage;