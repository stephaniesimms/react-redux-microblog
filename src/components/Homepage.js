import React, { Component } from 'react';
import PostList from './PostList';

class Homepage extends Component {
  render() {
    return (
      <PostList posts={this.props.posts}/>
    );
  }
}

export default Homepage;