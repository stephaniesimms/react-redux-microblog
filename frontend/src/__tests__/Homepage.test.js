import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Homepage from '../components/Homepage';

it('renders without crashing', function () {
  shallow(<Homepage />);
});

it('matches snapshot', function () {
  let wrapper = shallow(<Homepage />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});