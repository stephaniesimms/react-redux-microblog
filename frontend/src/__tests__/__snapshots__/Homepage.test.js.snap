// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<main>
  <p>
    Welcome to 
    <b>
      Microblog
    </b>
    , a fake forum for heavy metal-themed culinary delights. Here you will find diabolical dishes, forbidden fruits, and demonic desserts to exorcise your taste buds. Remember, gluttony is the best sin. Post your devil-icious dishes and leave comments. Vote for the posts you deem the most diabolical.
  </p>
  <Connect(TitleList) />
</main>
`;
